modules/AlphaWalletFoundation/AlphaWalletFoundation/Tokens/TokensAutodetector/TokensAutodetector.swift

```diff
@@ -36,27 +67,46 @@ public class SingleChainTokensAutodetector: NSObject, TokensAutodetector {
     private let detectedTokens: DetectedContractsProvideble
     private let tokensOrContractsDetectedSubject = PassthroughSubject<[TokenOrContract], Never>()
     private let getContractInteractions = GetContractInteractions()
-
+    private let contractToImportStorage: ContractToImportStorage
     public var tokensOrContractsDetected: AnyPublisher<[TokenOrContract], Never> {
         tokensOrContractsDetectedSubject.eraseToAnyPublisher()
     }
-    public var isAutoDetectingTransactedTokens = false
+    var isAutoDetectingTransactedTokens = false
     var isAutoDetectingTokens = false
 
     init(
             session: WalletSession,
+            contractToImportStorage: ContractToImportStorage,
             detectedTokens: DetectedContractsProvideble,
             withAutoDetectTransactedTokensQueue autoDetectTransactedTokensQueue: OperationQueue,
             withAutoDetectTokensQueue autoDetectTokensQueue: OperationQueue,
             importToken: ImportToken
     ) {
+        self.contractToImportStorage = contractToImportStorage
         self.importToken = importToken
         self.session = session
         self.detectedTokens = detectedTokens
         self.autoDetectTransactedTokensQueue = autoDetectTransactedTokensQueue
         self.autoDetectTokensQueue = autoDetectTokensQueue
     }
 
+    private static func loadContractsToDetect(server: RPCServer) -> [ContractToImport] {
+        guard let bundlePath = Bundle.main.path(forResource: "tokensToImport", ofType: "json") else { return [] }
```

Comment:
Shall we add a test for this `Bundle.main.path(forResource: "tokensToImport", ofType: "json")` ? Because it doesn't go through `R`, it could break silently.